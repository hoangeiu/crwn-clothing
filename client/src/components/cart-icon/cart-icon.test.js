import React from "react";
import { shallow } from "enzyme";

import { CartIcon } from "./cart-icon.component";

describe("CartIcon component", () => {
  let wrapper;
  let mocktToggleCartHidden;
  beforeEach(() => {
    mocktToggleCartHidden = jest.fn();

    const mockProps = {
      toggleCartHidden: mocktToggleCartHidden,
      itemCount: 0,
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it("should CartIcon render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call toggleCartHidden when icon is clicked", () => {
    wrapper.find(".cart-icon").simulate("click");
    expect(mocktToggleCartHidden).toHaveBeenCalled();
  });

  it("should render the itemCount as the text", () => {
    const itemCount = wrapper.find(".item-count").text();
    expect(itemCount.length).not.toBe(0);
  });
});
