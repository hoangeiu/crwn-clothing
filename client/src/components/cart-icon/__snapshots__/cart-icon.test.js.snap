// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should CartIcon render without error 1`] = `
<div
  className="cart-icon"
  onClick={[MockFunction]}
>
  <ForwardRef(SvgShoppingBag)
    className="shopping-icon"
  />
  <span
    className="item-count"
  >
    0
  </span>
</div>
`;
